parts/windows/kuberneteswindowssetup.ps1

```diff
@@ -487,6 +511,42 @@ try
         $timer.Stop()
         Write-Log -Message "We waited [$($timer.Elapsed.TotalSeconds)] seconds on NodeResetScriptTask"
     }
+
+    Write-Log "Stage2 completed successfully"
+    Logs-To-Event -TaskName "AKS.WindowsCSE.Stage2" -TaskMessage "Stage2 completed successfully"
+}
+
+try
+{
+    Logs-To-Event -TaskName "AKS.WindowsCSE.ExecuteCustomDataSetupScript" -TaskMessage ".\CustomDataSetupScript.ps1 -MasterIP $MasterIP -KubeDnsServiceIp $KubeDnsServiceIp -MasterFQDNPrefix $MasterFQDNPrefix -Location $Location -AADClientId $AADClientId -NetworkAPIVersion $NetworkAPIVersion -TargetEnvironment $TargetEnvironment -CSEResultFilePath $CSEResultFilePath"
+
+    # Exit early if the script has been executed
+    if (Test-Path -Path $CSEResultFilePath -PathType Leaf) {
+        Write-Log "The script has been executed before, will exit without doing anything."
+        return
+    }
+
+    # The provisioning is split into two stages to support VHD image creation workflows:
```

Comment:
It's great having this comment! Can you check to see if we strip comments before sending this CSE to the VM for provisioning? I'm constantly paranoid about CSE size.

Reply:
We probably don't do it for Windows (but seems like we have something for linux)